sdk/python/kfp/dsl/for_loop.py

```diff
@@ -39,8 +43,17 @@ def _get_loop_item_type(type_name: str) -> Optional[str]:
     Returns:
         The collection item type or None if no match found.
     """
-    match = re.match('(typing\.)?(?:\w+)(?:\[(?P<item_type>.+)\])', type_name)
-    return match['item_type'].lstrip().rstrip() if match else None
+    parameter_match = re.match('(typing\.)?(?:\w+)(?:\[(?P<item_type>.+)\])',
+                               type_name)
+    artifact_match = re.match('(system\.|google\.)(?P<type>\w+)@(\d\.\d\.\d)',
+                              type_name)
+
+    if parameter_match:
+        return parameter_match['item_type'].lstrip().rstrip()
+    elif artifact_match:
+        return type_name
+    else:
+        return None
```

Comment:
This seems like it may be duplicative of the changes to `type_utils.get_short_type_name`. Is there any opportunity to dedupe?

Separately, this comment may be obsolete due to other comments about how we can extract artifact type information.

Reply:
Deduped via `type_annotations.extract_artifact_subclass`